rusk-abi/src/lib.rs

```diff
@@ -63,10 +65,22 @@ const fn reserved(b: u8) -> ContractId {
     ContractId::from_bytes(bytes)
 }
 
-/// Generate a [`ContractId`] address from the given slice of bytes, that is
-/// also a valid [`BlsScalar`]
-pub fn gen_contract_id(bytes: &[u8]) -> ContractId {
-    ContractId::from_bytes(Hasher::digest(bytes).into())
+/// Generate a [`ContractId`] address from:
+/// - slice of bytes,
+/// - nonce
+/// - metadata
+/// that is also a valid [`BlsScalar`]
+pub fn gen_contract_id(
```

Comment:
`gen_contract_id` is in rusk-abi because it was there already before this PR, I only changed the implementation. I am open to suggestions about the proper location. We need to keep in mind that this functionality is also needed by clients such as deployment tool, etc.

Reply:
Since this is also needed for clients... @moCello what do you think? Should it go somewhere else, or remain where it is?

Reply:
The `ContractId` is a bit of a tricky type: currently we introduce it both in `piecrust-uplink` and `execution-core`. With issue #1859 we are drastically re-designing `rusk-abi` possibly moving functionalities to `execution-core` and `rusk`.
But moving functionalities around is not the scope of this PR so I tend to agree with Milosz that we should keep `gen_contract_id` where it is for now. I'll keep it in mind for the re-design though.